Combo PR (merge from NCAR/main, ECMWF GWD, maxmf bugfix, NEPTUNE reproducibility, mersenne bugfix)

grantfirl opened the pull request:
This PR combines many unrelated pieces of work. None of the changes are expected to change RT results.

- Updating from latest NCAR/main fork/branch. The following PRs have already been merged there.
  - uninitialized value bugfix in ugwp_driver_v0.F (credit @climbfuji ): https://github.com/NCAR/ccpp-physics/pull/1104
  - removal of gfdl_parse_tracers file in ccpp-physics (credit @climbfuji with host model (SCM and FV3) changes from @grantfirl); since GFS_typedefs.F90 compilation seems to be controlled through the CMakeLists.txt file in ccpp/physics, if we need to use the FMS version of get_tracer_index in GFS_typedefs, we need to add FMS as an included library in CMakeLists.txt for the FV3 only: https://github.com/NCAR/ccpp-physics/pull/1087
  - GNU -Wall warning reduction (credit @scrasmussen ): https://github.com/NCAR/ccpp-physics/pull/1105
  - Correction to #254 related to reading the NSSL MP-specific namelist (credit @grantfirl): https://github.com/NCAR/ccpp-physics/pull/1142 

- The following are new PRs into ufs/dev that are minor and don't change any RT results:
 - ECMWF non-stationary GWD (credit @BoYang-NOAA): https://github.com/ufs-community/ccpp-physics/pull/273
 - bugfix for maxMF (credit @SamuelTrahanNOAA ): https://github.com/ufs-community/ccpp-physics/pull/283
   - fixes https://github.com/ufs-community/ccpp-physics/issues/282
 - NEPTUNE reproducibility bugfix (credit @areinecke and @climbfuji )https://github.com/ufs-community/ccpp-physics/pull/290
   - fixes https://github.com/NCAR/ccpp-physics/issues/1140 
 - mersenne twister bugfix (credit @NickSzapiro-NOAA) : https://github.com/ufs-community/ccpp-physics/pull/292
 - compiler warning fixes (credit @NickSzapiro-NOAA): https://github.com/ufs-community/ccpp-physics/pull/297

scrasmussen committed:
Checking to define macro CCPP only if it is not defined reduces macro redefinition warnings

scrasmussen committed:
Fix 'Warning: CHARACTER expression will be truncated in assignment'

scrasmussen committed:
Fixed GNU warnings 'Warning: Nonconforming tab character at (1) [-Wtabs]'

scrasmussen committed:
Fixed GNU 'Warning: Line truncated at (1) [-Wline-truncation]' warnings

scrasmussen committed:
Fixed GNU warning 'Warning: Missing '&' in continued character constant at (1) [-Wampersand]'

scrasmussen committed:
Fixing 'Warning: Deleted feature: Start expression in DO loop at (1) must be integer' message by calling nint intrinsic. Copying the other do loops

scrasmussen committed:
Fix GNU warning "Warning: Return value err_message of function declared at (1) not set [-Wreturn-type]"

climbfuji committed:
Add physics/NRL_parse_tracers.F90 as a replacement for GFDL_parse_tracers for NEPTUNE

climbfuji committed:
Add missing tracers from NEPTUNE to physics/NRL_parse_tracers.F90

climbfuji committed:
Remove physics/NRL_parse_tracers.F90, physics/MP/GFDL/GFDL_parse_tracers.F90; update CODEOWNERS physics/docs/ccpp_doxyfile

dustinswales committed:
Optional argument cleanup: Radiation

dustinswales committed:
Optional argument cleanup: Convection

dustinswales committed:
Optional argument cleanup: GWD

dustinswales committed:
Optional argument cleanup: MP

dustinswales committed:
Optional argument cleanup: PBL

dustinswales committed:
Optional argument cleanup: Surface Layer

climbfuji committed:
Bug fix in physics/GWD/ugwp_driver_v0.F to prevent using uninitialized Bnv2 array in line 350

dustinswales committed:
Optional argument cleanup: Surface Models

dustinswales committed:
Optional argument cleanup: CICE and Smoke

dustinswales committed:
Optional argument cleanup: Some interstitials

dustinswales committed:
Revert removal of optional arguments in Interstitial schemes.

dustinswales committed:
REvert one more change

dustinswales committed:
Bug

dustinswales committed:
Anothe bug

dustinswales committed:
SOme more changes from RTs

scrasmussen committed:
Merge remote-tracking branch 'grant/rrfsv1-to-ufs/dev6' into ufs-dev-pr235

scrasmussen committed:
Merge pull request #1111 from scrasmussen/ufs-dev-pr235

UFS-dev PR#235

dustinswales committed:
Merge branch 'ufs/dev' of https://github.com/ufs-community/ccpp-physics into ufs-dev-pr251

dustinswales committed:
Merge pull request #1112 from dustinswales/ufs-dev-pr251

Ufs dev PR#251

dustinswales committed:
Revert some changes

dustinswales committed:
Revert merge conflict

scrasmussen committed:
Merge branch 'rrfsv1-to-ufs/dev10' into ufs-dev-pr252

dustinswales committed:
Merge pull request #1114 from scrasmussen/ufs-dev-pr252

UFS-dev PR#252 Tenth reconciliation PR from production/RRFS.v1

grantfirl committed:
Merge branch 'main' into bugfix/ugwp_driver_v0_uninit_main

grantfirl committed:
Merge pull request #1104 from climbfuji/bugfix/ugwp_driver_v0_uninit_main

Bug fix in physics/GWD/ugwp_driver_v0.F to prevent using uninitialized Bnv2 array in line 350

hertneky committed:
Merge branch 'conv_hr5update' into ufs-dev-PR232

hertneky committed:
Merge branch 'mr2_rst' into ufs-dev-PR232

climbfuji committed:
Merge branch 'main' of https://github.com/NCAR/ccpp-physics into feature/add_nrl_parse_tracers_ncar_main

dustinswales committed:
Merge pull request #1087 from climbfuji/feature/add_nrl_parse_tracers_ncar_main

Remove GFDL_parse_tracers.F90 (was: Add NRL_parse_tracers.F90 as a replacement for GFDL_parse_tracers.F90 for NEPTUNE)

grantfirl committed:
Merge pull request #1117 from hertneky/ufs-dev-PR232

UFS dev combined PR#232 and #256

hertneky committed:
Merge branch 'updraft_velocity' into ufs-dev-PR246

grantfirl committed:
fix intent error in GFS_suite_interstitial_3_run

grantfirl committed:
Merge pull request #1121 from hertneky/ufs-dev-PR246

Ufs dev pr246

grantfirl committed:
Merge branch 'feature/remove_gfs_int_radphys_reset' into ufs-dev-PR258

grantfirl committed:
Merge pull request #1127 from grantfirl/ufs-dev-PR258

UFS-dev PR#258

BoYang-NOAA committed:
add three ecmwf related files

scrasmussen committed:
Merge branch 'memcheck_fixes' into ufs-dev-PR260-memcheck-fixes

scrasmussen committed:
Merge branch 'feature/cmake_updates_from_neptune_20250318_for_ufs_dev' into ufs-dev-PR259-CMake-changes

scrasmussen committed:
hotfix: bumping Ubuntu version to latest, bumping python version to one used by modules

BoYang-NOAA committed:
Merge remote-tracking branch 'ufs-community/ufs/dev' into ecmwf

scrasmussen committed:
Merge pull request #1135 from scrasmussen/hotfix/bump-ci-ubuntu-version

CI Hotfix: bumping Ubuntu version to latest

scrasmussen committed:
Merge branch 'main' into ufs-dev-PR259-CMake-changes

scrasmussen committed:
Merge branch 'main' into ufs-dev-PR260-memcheck-fixes

scrasmussen committed:
Merge pull request #1105 from scrasmussen/gnu-wall-warning-reduction-part1

GNU -Wall Warning Reduction, Part 1

scrasmussen committed:
Moving pull_request_template up a directory so it will be used

scrasmussen committed:
PR and issue reference simplified and whitespace cleanup

BoYang-NOAA committed:
add a doc section in ecmwf_ngw_emc

scrasmussen committed:
Restrict Github actions CI mainly to NCAR's upstream and ufs-community's fork. This is so users don't get charged for CI

scrasmussen committed:
Merge pull request #1133 from scrasmussen/ufs-dev-PR260-memcheck-fixes

ufs-dev PR260 memcheck fixes

grantfirl committed:
Merge branch 'main' into ufs-dev-PR259-CMake-changes

grantfirl committed:
Merge pull request #1134 from scrasmussen/ufs-dev-PR259-CMake-changes

ufs-dev PR#259 CMake changes

grantfirl committed:
Merge branch 'main' into enhancement/restrict-ci-mainly-to-upstream

BoYang-NOAA committed:
revert scientific documentation

BoYang-NOAA committed:
add  copyright/Apache licence header

scrasmussen committed:
Merge pull request #1138 from scrasmussen/enhancement/restrict-ci-mainly-to-upstream

Restrict Github actions CI mainly to NCAR's upstream

hertneky committed:
revert accidental changes

hertneky committed:
another thing missed

SamuelTrahanNOAA committed:
bug fix to maxMF check: only access when do_mynnedmf=.true.

grantfirl committed:
Merge pull request #1139 from hertneky/ufs-dev-248

Ufs-dev PR248 - Loosen restriction on optional arguments

scrasmussen committed:
Merge pull request #1137 from scrasmussen/enhancement/pr-template

Enhancement: PR Template

hertneky committed:
Merge branch 'nssl3m-update2025a' into ufs-dev-PR254

climbfuji committed:
Change indices i3 and j3 in routine aer_property in radiation_aerosols.f to run from 1 instead of i1/j1 to fix a restart reproducibility issue with NEPTUNE

NickSzapiro-NOAA committed:
nrest=n+4 in mersenne_twister

grantfirl committed:
fix issue with namelist for NSSL MP

grantfirl committed:
update NSSL MP to use internal namelist like other physics in UFS

grantfirl committed:
add iostat to internal namelist read to capture read error for NSSL MP namelist

grantfirl committed:
Merge branch 'lakefrac_ice_sfcr' into ufs-dev-PR281

grantfirl committed:
Merge branch 'canopywater_glacier' into ufs-dev-PR278

grantfirl committed:
Merge pull request #1142 from grantfirl/ufs-dev-PR254

UFS-dev PR#254

grantfirl committed:
Merge branch 'main' into ufs-dev-PR281

grantfirl committed:
Merge branch 'main' into ufs-dev-PR278

grantfirl committed:
Merge branch 'gcycleupdate' into ufs-dev-PR278

grantfirl committed:
Merge pull request #1143 from grantfirl/ufs-dev-PR281

UFS-dev PR#281

grantfirl committed:
Merge branch 'main' into ufs-dev-PR278

grantfirl committed:
Merge pull request #1144 from grantfirl/ufs-dev-PR278

UFS-dev PR#278

grantfirl committed:
update CMakeLists.txt to use FMS for FV3 only

grantfirl committed:
Merge branch 'main' into combo_20250625

grantfirl committed:
Merge branch 'ufs/dev' into combo_20250625

grantfirl committed:
Merge branch 'ecmwf' into combo_20250625

grantfirl committed:
Merge branch 'bugfix/maxmf-check' into combo_20250625

grantfirl committed:
Merge branch 'bugfix/radiation_aerosols_restart_reproducibility_neptune' into combo_20250625

grantfirl committed:
Merge branch 'mersenne_nrest' into combo_20250625

grantfirl commented:
@MicroTed Here are the same changes that were merged into NCAR:main related to namelists if you'd like to take a look and review before it goes into ufs-community:ufs/dev

climbfuji reviewed (approved):
thanks!

rhaesung reviewed (approved):
Thanks, @grantfirl !

MicroTed commented:
@grantfirl It looks like it serves the current need. I may tweak it in a later update but no time for that right now.

NickSzapiro-NOAA committed:
tlim_okw instead of tlim_fgf in cires_ugwpv1_triggers.F90

tlim_fgf is defined above this in get_spectra_tau_nstgw but not in get_spectra_tau_okw

NickSzapiro-NOAA committed:
Initialize intent(out) zlwb in cires_ugwpv1_oro.F90

Fix warning of:
A dummy argument with an explicit INTENT(OUT) declaration is not given an explicit value.   [ZLWB]

NickSzapiro-NOAA committed:
Set jj if dlat > 0.0, dlon <= 0.0 in sfcsub.F

Fix Warning: 'jj' may be used uninitialized [-Wmaybe-uninitialized]

grantfirl committed:
Merge branch 'ufs/dev' into combo_20250625

grantfirl committed:
Merge remote-tracking branch 'nick-fork/warning_explicit_gwd' into combo_20250625

jkbk2004 commented:
All tests are done at https://github.com/ufs-community/ufs-weather-model/pull/2790. @rhaesung @dustinswales @grantfirl can you merge this pr?
